### PR TITLE
Drupal Views CiviCRM: Allow Participant record to be used from Event.

### PR DESCRIPTION
**Pull request by JohnFF**

### CODE REVIEW

**Comment by JohnFF** [comment]
@colemanw @eileenmcnaughton @seamuslee001 @jackrabbithanna < this is the new PR :) it has the changes suggested.

Please note that the civicrm_api check is still needed for the API call.

If someone else could please test this, and let Coleman know if you're happy for it to be merged, that would be amazeballs, and I'd happily return favour in future.

File: modules/views/civicrm/civicrm_handler_relationship_participant.inc [hunk 1]
```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * This relationship handler is used when joining the civicrm_relationship table
```
Comment (seamuslee001) [inline modules/views/civicrm/civicrm_handler_relationship_participant.inc]: @JohnFF are you able to update this code comment

File: modules/views/components/civicrm.core.inc [hunk 2]
```diff
@@ -157,6 +157,18 @@ function _civicrm_core_data(&$data, $enabled) {
...
 
+  $data['civicrm_event']['civicrm_participant'] = array(
+    'real field' => 'id',
+    'title' => t('CiviCRM Participant (starting from Event)'),
+    'help' => t('Allows a Participant Contact relationship from an Event'),
+    'relationship' => array(
+      'base' => 'civicrm_participant',
+      'base field' => 'event_id',
+      'handler' => 'civicrm_handler_relationship_participant',
+      'label' => t('CiviCRM Participant (starting from Event)'),
```
Comment (colemanw) [inline modules/views/components/civicrm.core.inc]: I don't understand this string (note that it appears twice, if we decide to change it).
What about "CiviCRM Event Participant"

Comment (JohnFF) [reply modules/views/components/civicrm.core.inc]: Hi @colemanw - this string appears twice in the same way that it does for CiviCRM Relationship. I also looked to "starting from Contact A" as guidance, although here I see it's probably not necessary. I've adjusted the PR accordingly.

**Comment by JohnFF** [comment]
image
image

**Comment by JohnFF** [comment]
You can check out the end results here: http://civicrm-drupal-john886170.codeanyapp.com (site may time out, I'm unsure atm).

**Comment by colemanw** [comment]
* ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain)) __Pass__
* ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test)) __Pass__
* ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code)) __Pass__
* ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc)) __Pass__
* ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint)) __Pass__
* ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run)) __Pass__: Tested on demo site
* ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user)) __Pass__: Meets user expectations for Views joins
* ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech)) __Pass__: consistent with other code.

**Pull request merged**